test/e2e/preset_test.go

```diff
@@ -351,6 +376,11 @@ func validateResourceStatus(workspaceObj *kaitov1beta1.Workspace) {
 				return false
 			}
 
+			workspaceJSON, err := json.Marshal(workspaceObj)
```

Comment:
This line is inside a `Eventually` function. It means it will be printed multiple times.
I guess you only want the spec definition for debugging. Make sure it will be printed only once.
Otherwise, there are too much verbose logs.

Reply:
One coding suggestion, save it and print it after `Eventually`

Reply:
Fei asked me to extract those changes to another PR